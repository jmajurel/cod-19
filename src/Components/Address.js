import React, { useState } from "react";
import Profile from "./Profile";

const Address = ({ existingAddress, onChange }) => {
  if (!existingAddress || existingAddress == {})
    existingAddress = {
      address1: "",
      address2: "",
      city: "",
      postCode: "",
      state: "",
      country: "",
    };
  const [address, setAddress] = useState(existingAddress);

  function handleOnChange() {
    onChange(address);
  }

  return (
    <div className="address">
      <div className="formGroup">
        <label>Address: </label>
        <input
          type="text"
          value={address.address1}
          onChange={(e) => {
            setAddress({ ...address, address1: e.target.value });
            onChange({ ...address, address1: e.target.value });
          }}
        />
      </div>
      <div className="formGroup">
        <label>Address Details: </label>
        <input
          type="text"
          value={address.address2}
          onChange={(e) => {
            setAddress({ ...address, address2: e.target.value });
            onChange({ ...address, address2: e.target.value });
          }}
        />
      </div>
      <div className="formGroup">
        <label>City: </label>
        <input
          type="text"
          value={address.city}
          onChange={(e) => {
            setAddress({ ...address, city: e.target.value });
            onChange({ ...address, city: e.target.value });
          }}
        />
      </div>
      <div className="formGroup">
        <label>Postcode: </label>
        <input
          type="text"
          value={address.postCode}
          onChange={(e) => {
            setAddress({ ...address, postCode: e.target.value });
            onChange({ ...address, postCode: e.target.value });
          }}
        />
      </div>
      <div className="formGroup">
        <label>State: </label>
        <input
          type="text"
          value={address.state}
          onChange={(e) => {
            setAddress({ ...address, state: e.target.value });
            onChange({ ...address, state: e.target.value });
          }}
        />
      </div>

      <div className="formGroup">
        <label>Country: </label>
        <input
          type="text"
          value={address.country}
          onChange={(e) => {
            setAddress({ ...address, country: e.target.value });
            onChange({ ...address, country: e.target.value });
          }}
        />
      </div>
    </div>
  );
};
export default Address;
