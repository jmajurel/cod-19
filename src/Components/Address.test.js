import React from "react";
import Address from "./Address";
import renderer from "react-test-renderer";
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";

let container = null;
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

jest.mock("react-i18next", () => ({
  // this mock makes sure any components using the translate hook can use it without a warning being shown
  useTranslation: () => {
    return {
      t: (str) => str,
      i18n: {
        changeLanguage: () => new Promise(() => {}),
      },
    };
  },
}));

test("display", () => {
  const component = renderer.create(<Address />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders with an address", () => {
  const mockAddress = { address1: "192 burnaby road" };
  act(() => {
    render(<Address existingAddress={mockAddress} />, container);
  });
  const label = container.querySelector("label");
  console.log("label.textContent: ", label.textContent);
  expect(label.textContent).toContain("Address");
});
