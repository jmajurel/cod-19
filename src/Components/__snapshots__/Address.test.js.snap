// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`display 1`] = `
<div
  className="address"
>
  <div
    className="formGroup"
  >
    <label>
      profile.address.address1Label
      : 
    </label>
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>
  <div
    className="formGroup"
  >
    <label>
      profile.address.address2Label
      : 
    </label>
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>
  <div
    className="formGroup"
  >
    <label>
      profile.address.cityLabel
      : 
    </label>
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>
  <div
    className="formGroup"
  >
    <label>
      profile.address.postCodeLabel
      : 
    </label>
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>
  <div
    className="formGroup"
  >
    <label>
      profile.address.stateLabel
      : 
    </label>
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>
  <div
    className="formGroup"
  >
    <label>
      profile.address.countryLabel
      : 
    </label>
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>
</div>
`;
